test/unit/legacy_wrappers/mongo_client.test.js

```diff
@@ -22,6 +24,54 @@ describe('legacy_wrappers/mongo_client.js', () => {
     await client.close();
   });
 
+  describe('client metadata', () => {
+    it('should set mongodb-legacy to the client metadata', () => {
```

Comment:
Can we structure these tests a bit better?  I'm thinking context blocks explaining the context for the test, followed by a better description ("set mongodb-legacy to the client metadata" isn't what happens, and isn't descriptive).  For example

```typescript
context('when no driverInfo is passed to the MongoClient constructor', () => {
  it('sets the driver name to mongodb-legacy', () => {});
  it(`sets the version to ${version}`, () => {}
});
```

I think this (or a similar structure) would apply to every new test added in this PR

Reply:
Organized!

Reply:
three comments

- the descriptions for the tests aren’t great.  for example - what does `should set mongodb-legacy to the client metadata` mean?  the test description should describe what the test is testing, `should assign client metadata.name to mongodb-legacy` .  this leads into the second comment
- the tests would be better organized with a single assertion per it block.  `should assign client metadata.name to mongodb-legacy and client metadata.version to the version in package.json` is a mouthful, but if it’s split into two separate it blocks, `should assign client metadata.name to mongodb-legacy` and `should assign client metadata.version to the version in the package.json` are both easy to understand and describe exactly what the test is testing
- the other tests would be much more understandable if we broke it block into a context + it (preferably two its actually).  for example

```typescript
it('should prepend mongodb-legacy to user passed driverInfo.name', () => {
        const client = new LegacyMongoClient(iLoveJs, { driverInfo: { name: 'mongoose' } });
        expect(client.options.metadata).to.have.nested.property(
          'driver.name',
          'nodejs|mongodb-legacy|mongoose'
        );
        expect(client.options.metadata)
          .to.have.property('version')
          .that.includes(currentLegacyVersion);
      });
```
would become
```typescript
context('when driverInto.name is provided and driverInfo.version is not', () => { 
  it('should prepend mongodb-legacy to user passed driverInfo.name', () => {...}
  it('should assign driverInfo.version to the version in the package.json', () => { ... }
});
```

Reply:
organized more!